proc_RunUpdateStatistics.sql

```diff
@@ -0,0 +1,313 @@
+   SELECT @v_EmailSubject = @@SERVERNAME + ' ' + @p_DatabaseName + ' Statistics Report';
+
+   -- Enabling debugging mode for additional details in email report
+   IF @p_IsDebug <> 1 AND @p_IsDebug <> 0
+      SET @p_IsDebug = 0
+   SET @v_QueriesExecuted = '';
+
+   -- Before doing anything, validate that the database exists.  This should
+   -- prevent any SQL Injection attacks
+   IF DB_ID( @p_DatabaseName ) IS NULL
+   BEGIN
+      SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'Database not found: ' + @p_DatabaseName;
+      GOTO done;
+   END;
+
+   IF @p_DaysSinceStatsUpdatedToForceUpdate < 0
+   BEGIN
+      SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'p_DaysSinceStatsUpdatedToForceUpdate must be greater than or equal to 0. Was ' + @p_DaysSinceStatsUpdatedToForceUpdate;
+      GOTO done;
+   END
+
+   SET @v_StaleStatisticsCutoffTime = DATEADD(DAY, -@p_DaysSinceStatsUpdatedToForceUpdate, GETDATE());
+
+   DECLARE @v_GetTablesCmd NVARCHAR(MAX);
+   SET @v_GetTablesCmd = '
+      SELECT
+         t.TABLE_SCHEMA AS SchemaName
+         ,t.TABLE_NAME AS TableName
+      FROM
+         ['+ @p_DatabaseName +'].INFORMATION_SCHEMA.TABLES AS t
+      WHERE
+         t.TABLE_TYPE = ''BASE TABLE''';
+
+   SET @v_OperationStartTime = GETDATE();
+
+   BEGIN TRY
+      INSERT INTO @v_DatabaseTablesTable EXECUTE sp_executesql @v_GetTablesCmd;
+   END TRY
+   BEGIN CATCH
+      SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'Failed to query database tables due to exception: ' + @v_NewLine + ERROR_MESSAGE();
+      GOTO done;
+   END CATCH;
+
+   SET @v_OperationStopTime = GETDATE();
+
+   SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'Finished querying for database tables. ' + 'Started At: ' + CAST( @v_OperationStartTime AS VARCHAR(20) ) + ' Ended At: ' + CAST( @v_OperationStopTime AS VARCHAR(20) ) + ' Took ' + CAST( DATEDIFF( SECOND, @v_OperationStartTime, @v_OperationStopTime ) AS VARCHAR(20) )  + ' seconds';
+   SELECT @v_QueriesExecuted = @v_QueriesExecuted + @v_NewLine + @v_NewLine + 'Get tables command: ' + @v_GetTablesCmd;
+
+   DECLARE
+       @v_CurCountSchema AS SYSNAME
+      ,@v_CurCountTable AS SYSNAME
+      ,@v_GetRowCountCmd AS NVARCHAR(MAX)
+      ,@v_RowCountCounter AS SMALLINT
+      ,@v_DatabaseTablesLastRow AS SMALLINT
+
+   SET @v_RowCountCounter = 1;
+   SELECT @v_DatabaseTablesLastRow = COUNT(1) FROM @v_DatabaseTablesTable;
+
+   SET @v_OperationStartTime = GETDATE();
+
+   -- Get current table row counts to compare to when stats were last updated
+   WHILE ( @v_RowCountCounter <= @v_DatabaseTablesLastRow )
+   BEGIN
+      SELECT
+          @v_CurCountSchema = d.SchemaName
+         ,@v_CurCountTable = d.TableName
+      FROM
+         @v_DatabaseTablesTable AS d
+      WHERE
+         d.DatabaseTableId = @v_RowCountCounter;
+
+      SET @v_GetRowCountCmd = '
+         SELECT
```

Comment:
Because Screencast has a fair amount of larger tables I think it would better to only query back the row counts of tables that have stale stats. Getting all these counts on live might take a while and cause a bit of churn.

Reply:
Actually since you're just targeting indexes, you could probably just use something like this: https://github.com/TechSmith/sql-server-index-maintenance/blob/main/proc_RunIndexMaintenance.sql#L151-L189. This will filter out any smaller indexes using the page count.